File: 38-count-and-say/count-and-say.java
class Solution {
    public String countAndSay(int n) {
        if (n <= 0) return "";
        
        String result = "1";
        
        for (int i = 1; i < n; i++) {
            StringBuilder current = new StringBuilder();
            int count = 1;
            
            for (int j = 0; j < result.length(); j++) {
                if (j + 1 < result.length() && result.charAt(j) == result.charAt(j + 1)) {
                    count++;
                } else {
                      current.append(count).append(result.charAt(j));
                    count = 1; 
                      }
            }
            
            result = current.toString();
        }
        
        return result;
    }
}

File: 38-count-and-say/count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n <= 0:
            return ""
        
        result = "1"
        
        for _ in range(1, n):
            current = []
            count = 1
            
            for j in range(len(result)):
                 if j + 1 < len(result) and result[j] == result[j + 1]:
                    count += 1
                 else:
                     current.append(str(count))
                     current.append(result[j])
                     count = 1 
            result = "".join(current)
            
        return result